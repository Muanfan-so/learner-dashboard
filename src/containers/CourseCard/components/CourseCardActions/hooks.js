import { handleEvent } from 'data/services/segment/utils';
import { eventNames } from 'data/services/segment/constants';

export const useCourseCardActionData = () => {
  const trackUpgradeClick = () => {
    handleEvent(eventNames.upgradeCourse, {
      pageName: 'learner_home',
      linkType: 'button',
      linkCategory: 'green_upgrade',
    });
  };

  return {
    trackUpgradeClick,
  };
};

export default useCourseCardActionData;
