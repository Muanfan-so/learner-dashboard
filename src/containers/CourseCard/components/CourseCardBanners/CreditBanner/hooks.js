import { StrictDict } from 'utils';
import { hooks as appHooks } from 'data/redux';

import ApprovedContent from './views/ApprovedContent';
import EligibleContent from './views/EligibleContent';
import MustRequestContent from './views/MustRequestContent';
import PendingContent from './views/PendingContent';
import RejectedContent from './views/RejectedContent';

export const statusComponents = StrictDict({
  pending: PendingContent,
  approved: ApprovedContent,
  rejected: RejectedContent,
});

export const useCreditBannerData = (cardId) => {
  const credit = appHooks.useCardCreditData(cardId);
  const { supportEmail } = appHooks.usePlatformSettingsData();
  if (!credit.isEligible) { return null; }

  const { error, purchased, requestStatus } = credit;
  let ContentComponent = EligibleContent;
  if (purchased) {
    if (requestStatus == null) {
      ContentComponent = MustRequestContent;
    } else if (Object.keys(statusComponents).includes(requestStatus)) {
      ContentComponent = statusComponents[requestStatus];
    }
    // Current behavior is to show Elligible State if unknown request status is returned
  }
  return {
    ContentComponent,
    error,
    supportEmail,
  };
};

export default {
  useCreditBannerData,
};
