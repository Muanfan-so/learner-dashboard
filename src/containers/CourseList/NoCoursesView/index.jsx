import React from 'react';
import { useIntl } from '@edx/frontend-platform/i18n';
import { Button, Image } from '@edx/paragon';
import { Search } from '@edx/paragon/icons';

import emptyCourseSVG from 'assets/empty-course.svg';

import { hooks as appHooks } from 'data/redux';
import messages from './messages';

import './index.scss';

export const NoCoursesView = () => {
  const { courseSearchUrl } = appHooks.usePlatformSettingsData();
  const { formatMessage } = useIntl();
  return (
    <div
      id="no-courses-content-view"
      className="d-flex align-items-center justify-content-center mb-4.5"
    >
      <Image src={emptyCourseSVG} alt={formatMessage(messages.bannerAlt)} />
      <h1>
        {formatMessage(messages.lookingForChallengePrompt)}
      </h1>
      <p>
        {formatMessage(messages.exploreCoursesPrompt)}
      </p>
      <Button
        variant="brand"
        as="a"
        href={courseSearchUrl}
        iconBefore={Search}
      >
        {formatMessage(messages.exploreCoursesButton)}
      </Button>
    </div>
  );
};

export default NoCoursesView;
