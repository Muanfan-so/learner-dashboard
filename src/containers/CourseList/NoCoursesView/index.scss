@import "@edx/paragon/scss/core/core";

#no-courses-content-view {
  border: 2px solid $light-400; // Can this be replaced by paragon?
  flex-direction: column;
  padding-bottom: map-get($spacers, 5);
  padding-top: map-get($spacers, 5);
  height: 100%; // Can this be replaced by paragon?

  & > * {
    margin-top: map-get($spacers, 3);
    margin-bottom: map-get($spacers, 3);
  }
}

