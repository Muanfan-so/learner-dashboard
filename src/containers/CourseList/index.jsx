import React from 'react';
import PropTypes from 'prop-types';

import CourseCard from 'containers/CourseCard';
import SelectSession from 'containers/SelectSession';

export const CourseList = ({ courseListData }) => (
  <div className="d-flex flex-column flex-grow-1">
    {courseListData.map((courseNumber) => (
      <CourseCard key={courseNumber} courseNumber={courseNumber} />
    ))}
    <SelectSession />
  </div>
);

CourseList.propTypes = {
  courseListData: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default CourseList;
