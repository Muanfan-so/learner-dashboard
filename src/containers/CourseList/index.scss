#course-list-heading-container {
  display: flex;
  justify-content: space-between;
}

.course-list-loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}