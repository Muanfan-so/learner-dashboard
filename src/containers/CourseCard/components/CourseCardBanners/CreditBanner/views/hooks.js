import React from 'react';

import { StrictDict } from 'utils';
import { apiHooks } from 'hooks';

import * as module from './hooks';

export const state = StrictDict({
  creditRequestData: (val) => React.useState(val), // eslint-disable-line
});

export const useCreditRequestData = (cardId) => {
  const [requestData, setRequestData] = module.state.creditRequestData(null);
  const createCreditApiRequest = apiHooks.useCreateCreditRequest(cardId);
  const createCreditRequest = (e) => {
    e.preventDefault();
    createCreditApiRequest().then(setRequestData);
  };
  return { requestData, createCreditRequest };
};

export default {
  useCreditRequestData,
};
