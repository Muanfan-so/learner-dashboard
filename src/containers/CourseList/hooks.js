import React from 'react';
import { useDispatch } from 'react-redux';

import { useCheckboxSetValues, useWindowSize, breakpoints } from '@edx/paragon';

import { StrictDict } from 'utils';
import { actions, hooks as appHooks } from 'data/redux';
import { ListPageSize, SortKeys } from 'data/constants/app';

import * as module from './hooks';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth;
};

export const state = StrictDict({
  sortBy: (val) => React.useState(val), // eslint-disable-line
});

export const useCourseListData = () => {
  const dispatch = useDispatch();
  const pageNumber = appHooks.usePageNumber();
  const [filters, setFilters] = useCheckboxSetValues([]);
  const [sortBy, setSortBy] = module.state.sortBy(SortKeys.enrolled);

  const { numPages, visible } = appHooks.useCurrentCourseList({
    sortBy,
    filters,
    pageSize: ListPageSize,
  });
  const handleRemoveFilter = (filter) => () => setFilters.remove(filter);
  const setPageNumber = (value) => dispatch(actions.app.setPageNumber(value));

  return {
    pageNumber,
    numPages,
    setPageNumber,
    visibleList: visible,
    filterOptions: {
      sortBy,
      setSortBy,
      filters,
      setFilters,
      handleRemoveFilter,
    },
    showFilters: filters.length > 0,
  };
};

export default useCourseListData;
