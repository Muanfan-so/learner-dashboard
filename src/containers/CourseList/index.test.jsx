import { shallow } from 'enzyme';

import CourseList from '.';
import { useCourseListData, useIsCollapsed } from './hooks';

jest.mock('./hooks', () => ({
  useCourseListData: jest.fn(),
  useIsCollapsed: jest.fn(),
}));

jest.mock('containers/CourseCard', () => 'CourseCard');
jest.mock('containers/CourseFilterControls', () => ({
  ActiveCourseFilters: 'ActiveCourseFilters',
  CourseFilterControls: 'CourseFilterControls',
}));

describe('CourseList', () => {
  const defaultCourseListData = {
    filterOptions: {},
    numPages: 1,
    setPageNumber: jest.fn().mockName('setPageNumber'),
    showFilters: false,
    visibleList: [],
  };
  useIsCollapsed.mockReturnValue(false);
  const createWrapper = (courseListData) => {
    useCourseListData.mockReturnValueOnce({
      ...defaultCourseListData,
      ...courseListData,
    });
    return shallow(<CourseList />);
  };

  describe('snapshots', () => {
    test('with no filters', () => {
      const wrapper = createWrapper();
      expect(wrapper).toMatchSnapshot();
    });
    test('with filters', () => {
      const wrapper = createWrapper({
        filterOptions: {
          abitary: 'filter',
        },
        showFilters: true,
      });
      expect(wrapper).toMatchSnapshot();
    });
    test('with multiple courses and pages', () => {
      const wrapper = createWrapper({
        visibleList: [{ cardId: 'foo' }, { cardId: 'bar' }, { cardId: 'baz' }],
        numPages: 3,
      });
      expect(wrapper).toMatchSnapshot();
    });
    test('collapsed with multiple courses and pages', () => {
      useIsCollapsed.mockReturnValueOnce(true);
      const wrapper = createWrapper({
        visibleList: [{ cardId: 'foo' }, { cardId: 'bar' }, { cardId: 'baz' }],
        numPages: 3,
        showFilters: true,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
