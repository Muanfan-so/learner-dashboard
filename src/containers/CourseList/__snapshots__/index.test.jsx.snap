// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList collapsed with multiple courses and pages snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls />
    </div>
  </div>
  <List
    filterOptions={Object {}}
    numPages={3}
    setPageNumber={[MockFunction setPageNumber]}
    showFilters={true}
    visibleList={
      Array [
        Object {
          "cardId": "foo",
        },
        Object {
          "cardId": "bar",
        },
        Object {
          "cardId": "baz",
        },
      ]
    }
  />
</div>
`;

exports[`CourseList no courses snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls />
    </div>
  </div>
  <List
    filterOptions={Object {}}
    numPages={1}
    setPageNumber={[MockFunction setPageNumber]}
    showFilters={false}
    visibleList={Array []}
  />
</div>
`;

exports[`CourseList no filters snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls />
    </div>
  </div>
  <List
    filterOptions={Object {}}
    numPages={1}
    setPageNumber={[MockFunction setPageNumber]}
    showFilters={false}
    visibleList={Array []}
  />
</div>
`;

exports[`CourseList with filters snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls
        abitary="filter"
      />
    </div>
  </div>
  <List
    filterOptions={
      Object {
        "abitary": "filter",
      }
    }
    numPages={1}
    setPageNumber={[MockFunction setPageNumber]}
    showFilters={true}
    visibleList={Array []}
  />
</div>
`;

exports[`CourseList with multiple courses and pages snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls />
    </div>
  </div>
  <List
    filterOptions={Object {}}
    numPages={3}
    setPageNumber={[MockFunction setPageNumber]}
    showFilters={false}
    visibleList={
      Array [
        Object {
          "cardId": "foo",
        },
        Object {
          "cardId": "bar",
        },
        Object {
          "cardId": "baz",
        },
      ]
    }
  />
</div>
`;
